test/gurka/test_turn_lanes.cc

```diff
@@ -43,3 +59,240 @@ TEST(Standalone, TurnLanes) {
+                                  {0, ""},
+                              });
+
+  // A -> E - takes the leftmost through lane
+  result = gurka::route(map, "A", "E", "auto");
+  validate_turn_lanes(result, {
+                                  {3, "[ left | left;through ACTIVE | through;right ]"},
+                                  {0, ""}, // TODO lanes are tossed when all are through
+                              });
+
+  // A -> G - takes the rightmost through lane
+  result = gurka::route(map, "A", "G", "auto");
+  gurka::assert::osrm::expect_steps(result, {"Bronx and Pelham Parkway", "Pelham Parkway South"});
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kSlightRight,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+  validate_turn_lanes(result, {
+                                  {3, "[ left | left;through | through;right ACTIVE ]"},
+                                  {0, ""}, // TODO lanes are tossed when all are through
+                              });
+
+  // A -> H - takes the right lane
+  result = gurka::route(map, "A", "H", "auto");
+  validate_turn_lanes(result, {
+                                  {3, "[ left | left;through | through;right ACTIVE ]"},
+                                  {0, ""},
+                              });
+}
+
+// Shared turn lane example
+// (40.053550, -76.309125)
+TEST(Standalone, TurnLanesSharedTurnLane) {
+  const std::string ascii_map = R"(
+    A---B---D---E---G
+        |       |
+        C       F
+  )";
+
+  const gurka::ways ways = {{"AB",
+                             {{"highway", "primary"},
+                              {"lanes:forward", "3"},
+                              {"turn:lanes:forward", "through|through|right"}}},
+                            {"BC", {{"highway", "residential"}}},
+                            {"BD",
+                             {{"highway", "primary"},
+                              {"lanes:forward", "3"},
+                              {"turn:lanes:forward", "through|through|right"}}},
+                            {"DE",
+                             {{"highway", "primary"},
+                              {"lanes:forward", "3"},
+                              {"turn:lanes:forward", "through|through|right"}}},
+                            {"EF", {{"highway", "residential"}}},
+                            {"EG",
+                             {{"highway", "primary"},
+                              {"lanes:forward", "2"},
+                              {"turn:lanes:forward", "through|through"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_turn_lanes_3");
+
+  // A -> G  - only through lanes should be active throughout
+  auto result = gurka::route(map, "A", "C", "auto");
```

Comment:
Looks like this route should be `A->G`? `A->C` is repeated otherwise.

Reply:
Good catch! Fixed in https://github.com/valhalla/valhalla/pull/2613/commits/d83764a10d2bade40134078f6a364e17d0875a7d